calculator.js

```diff
@@ -0,0 +1,150 @@
+/* global math */
+// Variables
+var buttons = [['7','8','9','+'],['4','5','6','-'],['1','2','3','*'],['0','.','=','/']];
+var theme = 'dark';
+var lastNum = '=';
+
+function prep() {
+  // Create base elements for the page
+  var container = document.createElement('div');
+  var header = document.createElement('div');
+  var nav = document.createElement('div');
+  var main = document.createElement('div');
+  var aside = document.createElement('div');
+  var footer = document.createElement('div');
+
+  // Apply IDs to the base elements
+  container.id = 'container';
+  header.id = 'header';
+  nav.id = 'nav';
+  main.id = 'main';
+  aside.id = 'aside';
+  footer.id = 'footer';
+
+  // Append main elements to container
+  container.appendChild(header);
+  container.appendChild(nav);
+  container.appendChild(main);
+  container.appendChild(aside);
+  container.appendChild(footer);
+
+  // Add container to page
+  document.body.appendChild(container);
+
+  heading(header);
+  calculator(main);
+  themeButtons(footer);
+
+  var about = document.createElement('p');
+  about.appendChild(document.createTextNode('Created by keawade'));
+  footer.appendChild(about);
+}
+
+function heading(head) {
+  // Add heading to header
+  var title = document.createElement('h1');
+  title.className = 'title';
+  title.textContent = 'Calculon';
+  head.appendChild(title);
+}
+
+function calculator(base) {
+  // Add calculator wrapper div
+  var calculatorWrap = document.createElement('div');
+  calculatorWrap.id = 'calculator';
+  base.appendChild(calculatorWrap);
+
+  // Add input box
+  var box = document.createElement('input');
+  box.className = 'cal-box';
+  box.id = 'cal-box';
+  calculatorWrap.appendChild(box);
+
+  // Add calculator buttons
+  for(var i = 0; i < buttons.length; i++) {
+    for(var j = 0; j < buttons[i].length; j++) {
+      var buttonContent = buttons[i][j];
+      var button = document.createElement('button');
+      button.className = 'cal-button';
+      button.id = buttonContent;
+      button.appendChild(document.createTextNode(buttonContent));
+      button.addEventListener('click', handleButton, false);
+      calculatorWrap.appendChild(button);
+    }
+  }
+  // If Enter/Return is pressed, click '=' to evaluate
+  document.addEventListener('keypress', function(key){
+    console.log(String.fromCharCode(key.charCode));
+    if(key.keyCode == 13){
+      document.getElementById('=').click();
+    } else if (/[0-9]|[/*-+=.]/.test(String.fromCharCode(key.charCode))) {
+      document.getElementById(String.fromCharCode(key.charCode)).click();
```

Comment:
Variable are handy (and performant!) alternative when you find yourself using the same code `String.fromCharCode(key.charCode)` in three different places.